src/nunit.analyzers/Helpers/CodeFixHelper.cs

```diff
@@ -35,6 +35,39 @@ internal static class CodeFixHelper
             return invocationExpression.ReplaceNode(invocationTargetNode, newInvocationTargetNode);
         }
 
+        /// <summary>
+        /// This is assumed to be arguments for an 'Assert.That(actual, constraint, "...: {0} - {1}", param0, param1)`
+        /// which needs converting into 'Assert.That(actual, constraint, $"...: {param0} - {param1}").
+        /// </summary>
+        /// <param name="arguments">The arguments passed to the 'Assert' method. </param>
+        /// <param name="minimumNumberOfArguments">The argument needed for the actual method, any more are assumed messages.</param>
+        public static ArgumentSyntax? GetInterpolatedMessageArgumentOrDefault(ArgumentSyntax? messageArgument, List<ArgumentSyntax> args)
+        {
+            if (messageArgument is null)
+                return null;
+
+            var formatSpecificationArgument = messageArgument.Expression;
+            if (formatSpecificationArgument.IsKind(SyntaxKind.NullLiteralExpression))
+                return null;
+
+            // We only support converting if the format specification is a constant string.
+            if (args.Count == 0 || formatSpecificationArgument is not LiteralExpressionSyntax literalExpression)
+                return messageArgument;
+
+            var formatSpecification = literalExpression.Token.ValueText;
+
+            var argsExpression = args.Single(a => a != messageArgument).Expression;
```

Comment:
Need to fix this logic

Reply:
I don't follow this logic:
1.  _messageArgument_ is not be part of _args_ anymore.
2. `Single` implies only one argument. This will fail in some case where named arguments are used *in order*:
`ClassicAssert.AreEqual(expected: 2.0, actual: 3.0, delta: 0.1, message: "Named parameters in order {0}, {1}", "first", "second");`
4.  The compiler doesn't allow unnamed argument when out of order parameters are used.  In that case _args_ would not have to be an array and this code should create `() => string.Format(messageArgument, args)` as per #720